account-kit/smart-contracts/src/ma-v2/client/client.test.ts

```diff
@@ -127,6 +142,118 @@ describe("MA v2 Tests", async () => {
+    });
+
+    // send UO with webauthn gas estimator
+    const builtUO = await provider.buildUserOperation({
+      uo: {
+        target: target,
+        value: sendAmount,
+        data: "0x",
+      },
+    });
+
+    const request = deepHexlify(builtUO);
+    if (!isValidRequest(request)) {
+      throw new InvalidUserOperationError(builtUO);
+    }
+
+    const uoHash = provider.account
+      .getEntryPoint()
+      .getUserOperationHash(request);
+
+    console.log("uoHash", uoHash);
+
+    let signedUOHash = await provider.account.signUserOperationHash(uoHash);
+
+    const signedUO = await provider.signUserOperation({ uoStruct: builtUO });
+
+    signedUO.signature = signedUOHash;
+
+    await provider.sendRawUserOperation(
+      signedUO,
+      provider.account.getEntryPoint().address,
+    );
+  });
+
+  it.fails(
+    "successfully sign + validate a message, for WebAuthn account",
+    async () => {
+      const provider = await givenWebAuthnProvider();
+
+      await setBalance(instance.getClient(), {
+        address: provider.getAddress(),
+        value: parseEther("2"),
+      });
+
+      const message = "0xdeadbeef";
+
+      let signature = await provider.signMessage({ message });
+
+      const publicClient = instance.getClient().extend(publicActions);
+
+      const domain = {
+        name: keccak256(stringToBytes("MyDapp")),
+        version: keccak256(stringToBytes("1")),
+        chainId: provider.chain.id,
+        verifyingContract: getDefaultWebauthnValidationModuleAddress(
+          provider.chain,
+        ),
+      };
+
+      const typeHash = keccak256(
+        //ethers.utils.toUtf8Bytes
+        stringToBytes(
+          "EIP712Domain(string name,string version,number chainId,address verifyingContract)",
+        ),
+      );
+
+      // TODO: consolidate formatting message to EIP-712 format (https://github.dev/fractional-company/contracts/blob/master/src/OpenZeppelin/drafts/EIP712.sol) with 1271 validation as part of the format + prepare signature work
+      const messageEIP712 = keccak256(
+        encodePacked(
+          ["bytes32", "bytes32", "bytes32"],
+          [
+            keccak256(stringToBytes("\x19\x01")),
+            keccak256(
+              encodeAbiParameters(
+                [
+                  { name: "typeHash", type: "bytes32" },
+                  { name: "name", type: "bytes32" },
+                  { name: "version", type: "bytes32" },
+                  { name: "chainId", type: "uint256" },
+                  { name: "verifyingContract", type: "address" },
+                ],
+                [
+                  typeHash,
+                  domain.name,
+                  domain.version,
+                  BigInt(domain.chainId),
+                  domain.verifyingContract,
+                ],
+              ),
+            ),
+            keccak256(stringToBytes(message)),
+          ],
+        ),
+      );
+
+      const isValid = await publicClient.verifyMessage({
```

Comment:
let's leave a comment that this should actually be `verifyTypedData` if we plan on fixing it later